.tekton/release/bundle-update/pipeline.yaml

```diff
@@ -0,0 +1,163 @@
+          - name: forkGitUrl
+            type: string
+            description: The git repository URL where the updated catalog is committed to
+            default: "https://github.com/raptorsun/lightspeed-operator.git"
+          - name: forkBranch
+            type: string
+            description: The branch of the git repository where the updated catalog is committed to
+            default: "bundle-update"
+          - name: forkUser
+            type: string
+            description: The Github user of the fork git repository
+            default: "raptorsun"
+          - name: sourceGitUrl
+            type: string
+            description: The source git repository URL to clone as starting point for the catalog update
+            default: "https://github.com/openshift/lightspeed-operator.git"
+          - name: sourceBranch
+            type: string
+            description: The branch of the source git repository to clone as starting point for the catalog update
+            default: "main"
+        steps:
+          - name: git-clone
+            image: quay.io/hasun/bundle-update:0.0.1
+            script: |
+              git clone --single-branch --branch $(params.sourceBranch) $(params.sourceGitUrl) repo
+              cd repo
+              git checkout -b $(params.forkBranch)
+            workingDir: /workspace
+          - name: update-file
+            image: quay.io/hasun/bundle-update:0.0.1
+            script: |
+              set -e
+              oc login --token=${KONFLUX_TOKEN} --server=https://api.stone-prd-rh01.pg1f.p1.openshiftapps.com:6443
+              IFS='/' read -r RELEASE_NAMESPACE RELEASE_NAME <<< "$(params.release)"
+              SNAPSHOT_IMAGES=$(oc get snapshot "$(params.snapshot)" -n "$RELEASE_NAMESPACE" \
+              -o jsonpath='{.spec.components}')
+              echo "Snapshot images: $SNAPSHOT_IMAGES"
+              cd repo
+              # back up the bundle element
+              jq '.[] | select(.name == "lightspeed-operator-bundle")' related_images.json > bundle.json
+              ./hack/snapshot_to_image_list.sh -s "$(params.snapshot)" -r stable -o related_images.json
+              # restore the bundle element
+              jq 'map(select(.name != "lightspeed-operator-bundle"))' related_images.json > tmp.json && mv tmp.json related_images.json
+              jq -s '.[0] + [.[1]]' related_images.json bundle.json > tmp.json && mv tmp.json related_images.json
+            workingDir: /workspace
+            env:
+              - name: KONFLUX_TOKEN
+                valueFrom:
+                  secretKeyRef:
+                    name: release-bot-konflux-token
+                    key: konflux-token
+              - name: HOME
+                value: /workspace
+          - name: git-config
+            image: quay.io/hasun/bundle-update:0.0.1
+            script: |
+              cd repo
+              # todo: replace the email and name with the bot's email and name
```

Comment:
Was this todo before or after merging to main?

Reply:
yes, when the pipeline is fully test on konflux, I will make a team account as bot account and switch to it